test/simple_operations.js

```diff
@@ -1,15 +1,355 @@
 "use strict";
 
-let SimpleOperations = artifacts.require("./SimpleOperations.sol");
+const { step } = require("mocha-steps");
+
+const SimpleOperations = artifacts.require("./SimpleOperations.sol");
 
 contract("SimpleOperations", accounts => {
   it("should initialize the contract with the parity client", async () => {
-    let operations = await SimpleOperations.deployed();
-    let owner = await operations.client.call("parity");
+    const operations = await SimpleOperations.deployed();
+    const owner = await operations.client("parity");
+
+    assert.equal(owner, accounts[0]);
+
+    // the creator of the operations contract should be set as the owner of the parity client
+    const client = await operations.clientOwner(accounts[0]);
+    assert.equal(web3.toUtf8(client), "parity");
+  });
+
+  it("should emit a `Received` event on fallback", async () => {
+    const operations = await SimpleOperations.deployed();
+    const watcher = operations.Received();
+
+    await operations.sendTransaction({
+      from: accounts[1],
+      value: 3,
+      data: web3.fromUtf8("hello"),
+    });
+
+    const events = await watcher.get();
+
+    assert.equal(events.length, 1);
+    assert.equal(events[0].args.from, accounts[1]);
+    assert.equal(events[0].args.value.valueOf(), 3);
+    assert.equal(web3.toUtf8(events[0].args.data), "hello");
+  });
 
+  it("should allow the owner of a client to transfer ownership", async () => {
+    const operations = await SimpleOperations.new();
+    const watcher = operations.ClientOwnerChanged();
+
+    // only the owner of the client can transfer ownership
+    try {
+      await operations.setClientOwner(accounts[2], { from: accounts[1] });
```

Comment:
In case `setClientOwner` suceeds we should actually fail the test, we don't do that here.
Also might be a good idea to abstract this pattern to a function, cause it's used often.

Reply:
Yep, I realized that and fixed it in PR #4 (https://github.com/parity-contracts/auto-updater/pull/4/commits/a95fada78af911f2fa315d7e463a4e52c9fc5790).